### PR TITLE
DRA: API cleanup + fix

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

This is a minor documentation and OpenAPI update. The ResourceHandle.DriverName was already required
via validation, but was not declared as such by the OpenAPI.

#### Does this PR introduce a user-facing change?
```release-note
DRA: client-side validation of a ResourceHandle would have accepted a missing DriverName, whereas server-side validation then would have raised an error.
```

### CODE REVIEW

File: pkg/apis/resource/namedresources.go [hunk 1]
```diff
@@ -92,8 +92,16 @@ type NamedResourcesRequest struct {
 
 // NamedResourcesFilter is used in ResourceFilterModel.
 type NamedResourcesFilter struct {
-	// Selector is a selector like the one in Request. It must be true for
-	// a resource instance to be suitable for a claim using the class.
+	// Selector is a CEL expression which must evaluate to true if a
+	// resource instance is suitable. The language is as defined in
+	// https://kubernetes.io/docs/reference/using-api/cel/
+	//
+	// In addition, for each type NamedResourcesin AttributeValue there is a map that
```
Comment (thockin) [inline pkg/apis/resource/namedresources.go:99]: "NamedResourcesin" ?

Comment (pohly) [reply pkg/apis/resource/namedresources.go:99]: /lgtm cancel

Will fix. How ironic that a PR that fixes typos introduces new ones...

Comment (pohly) [reply pkg/apis/resource/namedresources.go]: Fixed: https://github.com/kubernetes/kubernetes/compare/20988de8cca8f7608987907a13c97767e1421ccb..acfaf8c27590c78fe9900f8c7c1e99927392c20e

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**